ui/src/Credentials.ts

```diff
@@ -41,3 +42,49 @@ export const computeCredentials = (displayName: string): Credentials => {
       : `${httpBaseUrl}${displayName}/`
   };
 }
+
+type InternalPartyDetails = {
+  partyId : string,
+  token : string,
+  ledgerId : string,
+}
+
+const getDetails = (displayName : string): InternalPartyDetails => {
+  const uploadedCredentialsMap = retrieveParties();
+  const credentialsMap =
+    uploadedCredentialsMap === undefined ?
+    preloadedCredentialsMap :
+    uploadedCredentialsMap;
+  const {party, partyId, token, ledgerId } = credentialsMap.find(p => p.partyName === displayName)!;
+  // Either of those fields exists. This is for backward compatibility.
+  const partyIdentifier = party || partyId;
+  return { partyId: partyIdentifier, token, ledgerId }
+}
+
+const PARTIES_STORAGE_KEY = 'imported_parties';
+
+export function storeParties(parties: PartyDetails[]): void {
+    localStorage.setItem(PARTIES_STORAGE_KEY, JSON.stringify(parties));
+}
+
+export function retrieveParties(validateParties: boolean = true): PartyDetails[] | undefined {
+    const partiesJson = localStorage.getItem(PARTIES_STORAGE_KEY);
+
+    if (!partiesJson) {
+        return undefined;
+    }
+
+    const [ parties, error ] = convertPartiesJson(partiesJson, ledgerId, validateParties);
+
+    if (error) {
+        console.warn("Tried to load an invalid parties file from cache.", error);
```

Comment:
wondering if we want to tolerate these errors and not throw an exception instead?

Reply:
done